### ISSUE TITLE
Notarize macOS app bundle for launcher

### ISSUE THREAD

**Issue by rinrincodes** [Mica-Technologies/Minecraft-Forge-Launcher#2]
This will be a new requirement by Apple starting in 2020 where all apps must be notarized and have a valid and signed signature to be safe from malicious code and tampering.

**Comment by hawka97** [comment]
This will be completed in the future. This issue will remain open and will be revisited soon.

**Comment by hawka97** [comment]
relates to #25

**Comment by rinrincodes** [comment]
We still need to notarize the app bundle on or before January 1st, 2020. We still have time to do this but the sooner the better would be preferred and to prevent later issues with the launcher not being able to run on macOS later on.

**Comment by hawka97** [comment]
As far as I know, the strict requirements came in effect with Catalina, and won't change in 2020. They'll stay the same as they are now until this issue is completed (which may be sooner than 2020).

**Comment by rinrincodes** [comment]
Sadly it will not be possible till I get a paid developer account with Apple and make Mica Technologies a development team with them one day in the near future. When I do feel free to reopen this issue.

**Comment by rinrincodes** [comment]
Reopening for it to be added into 2020.2 later on.

**Comment by hawka97** [comment]
I've completed this earlier than expected. Two scripts have been added in the deploy folder, one for codesigning and putting the app into DMG form. The other is for submitting the dmg for notarization, checking for notarization status, and when notarization is done and passes Apple, the script can staple the notarization to the dmg. NOTE: These scripts cannot be used by persons outside of Mica Technologies, without modification to change the Apple Developer account email.
